locator/guice/src/main/java/org/eclipse/kapua/locator/guice/GuiceLocatorImpl.java

```diff
@@ -129,230 +133,64 @@ private void init(String locatorConfigName) throws Exception {
         //
         // Scan packages listed in to find KapuaModules
-        Collection<String> packageNames = locatorConfig.getIncludedPackageNames();
-        Reflections reflections = new Reflections(packageNames);
-        Set<Class<? extends AbstractKapuaModule>> kapuaModuleClasses = reflections.getSubTypesOf(AbstractKapuaModule.class);
+        final Reflections reflections = new Reflections(locatorConfig.getIncludedPackageNames());
+        final Set<Class<? extends AbstractKapuaModule>> kapuaModuleClasses = reflections.getSubTypesOf(AbstractKapuaModule.class);
 
         //
         // Instantiate Kapua modules
-        List<AbstractKapuaModule> kapuaModules = new ArrayList<>();
-        List<Class<? extends AbstractKapuaModule>> excludedKapuaModules = new ArrayList<>();
-        for (Class<? extends AbstractKapuaModule> moduleClazz : kapuaModuleClasses) {
-            if (isExcluded(moduleClazz.getName(), locatorConfig.getExcludedPackageNames())) {
-                excludedKapuaModules.add(moduleClazz);
-                continue;
-            }
+        final Map<Boolean, List<Class<? extends AbstractKapuaModule>>> includedAndExcludedModules = kapuaModuleClasses
+                .stream()
+                .collect(Collectors.partitioningBy(moduleClazz -> !locatorConfig.getExcludedPackageNames().stream().anyMatch(pn -> moduleClazz.getName().startsWith(pn))));
 
-            kapuaModules.add(moduleClazz.newInstance());
+        final List<AbstractKapuaModule> instantiatedKapuaModules = new ArrayList<>();
+        for (Class<? extends AbstractKapuaModule> moduleClazz : includedAndExcludedModules.get(true)) {
+            instantiatedKapuaModules.add(moduleClazz.newInstance());
         }
 
         // KapuaModule will be removed as soon as bindings will be moved to local modules
-        kapuaModules.add(new KapuaModule(locatorConfigName));
-
-        //
-        // Print loaded stuff
-        printLoadedKapuaModuleConfiguration(locatorConfigURL, locatorConfig, kapuaModules, excludedKapuaModules);
-
-        //
-        // Scan XmlSerializable
-        Set<Class<?>> xmlSerializableClasses = reflections.getTypesAnnotatedWith(XmlRootElement.class);
-        List<Class<?>> loadedXmlSerializables = new ArrayList<>();
-        List<Class<?>> excludedXmlSerializables = new ArrayList<>();
-        for (Class<?> xmlSerializableClass : xmlSerializableClasses) {
-            if (isExcluded(xmlSerializableClass.getName(), locatorConfig.getExcludedPackageNames()) ||
-                    !xmlSerializableClass.isAnnotationPresent(XmlRootElement.class)) {
-                excludedXmlSerializables.add(xmlSerializableClass);
-                continue;
-            }
-            loadedXmlSerializables.add(xmlSerializableClass);
-        }
-        ServiceModuleJaxbClassConfig.setSerializables(loadedXmlSerializables);
+        instantiatedKapuaModules.add(new KapuaModule(locatorConfigName));
```

Comment:
NEVER user static methods to pass around data, where dependency injection can be used